packages/pwa/app/pwa-kit.config.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import path from 'path'
+import Config from 'pwa-kit-react-sdk/config'
+
+// The Config class automatically validate the config object.
+// Error will be thrown if configuration object is invalid.
+export default new Config({
```

Comment:
The only one configuration file approach can end up generating a large and complex configuration file. 
- Some of the settings are unrelated to others. 
- We don't want developers to change some of the default settings because they will end with a broken build. 
- Developers can easily break things when they need to change a value manually.

One potential solution is to use a configuration folder instead and split the configuration details that we want developers to change into small configuration files.